src/java/org/apache/cassandra/utils/binlog/BinLog.java

```diff
@@ -462,36 +468,61 @@ public BinLog build(boolean cleanDirectory)
         }
     }
 
+    /**
+     * ChronicleQueue fails to start on cq4 files which are empty. Find such files in log dir and remove them.
+     */
+    private static Throwable cleanEmptyLogFiles(File directory, Throwable accumulate)
+    {
+        return cleanDir(directory, accumulate,
+                        (dir) -> dir.listFiles(file -> !file.isDirectory()
+                                                       && file.length() == 0
+                                                       && file.getName().endsWith(SingleChronicleQueue.SUFFIX)));
+    }
+
     public static Throwable cleanDirectory(File directory, Throwable accumulate)
     {
-        if (!directory.exists())
-        {
-            return Throwables.merge(accumulate, new RuntimeException(String.format("%s does not exists", directory)));
-        }
-        if (!directory.isDirectory())
-        {
-            return Throwables.merge(accumulate, new RuntimeException(String.format("%s is not a directory", directory)));
-        }
-        for (File f : directory.listFiles())
-        {
-            accumulate = deleteRecursively(f, accumulate);
-        }
+        return cleanDir(directory, accumulate, File::listFiles);
+    }
+
+    private static Throwable cleanDir(File directory, Throwable accumulate, Function<File, File[]> lister)
```

Comment:
nit: Would it be clearer just to keep the name `cleanDirectory()`? (i.e. It's an overload, and shortening the name doesn't really buy us anything.)